### ISSUE TITLE
Invalid Free causes parsing the valgrind output to fail

### ISSUE THREAD

**Issue by jfrimmel** [jfrimmel/cargo-valgrind#20]
If you have an invalid free, `cargo-valgrind` currently exits with an error. This is due to a changed Valgrind XML:
```xml
<error>
  <unique>...</unique>
  <tid>...</tid>
  <kind>InvalidFree</kind>
  <what>Invalid free() / delete / delete[] / realloc()</what>
  <stack>...</stack>
  <auxwhat>Address 0x8c2b840 is 0 bytes inside a block of size 16 free'd</auxwhat>
  <stack>...</stack>
  <auxwhat>Block was alloc'd at</auxwhat>
  <stack>...</stack>
</error>
```
Note the multiple stack fields.

To resolve this error the stack [`Error` structure](https://github.com/jfrimmel/cargo-valgrind/blob/82940566b2ea6a1b2dfcf918b3bebc2c2cda3d3d/src/valgrind_xml/mod.rs#L54-L62) has to be changed, so that it provides a vector of `Stack`s.
Since the `valgrind_xml` module is not public, the change won't not API breaking.

**Comment by jfrimmel** [comment]
At this point, the `<auxwhat>`-block should also be parsed and maybe presented to the user.

**Comment by jfrimmel** [comment]
This requires to parse either a `Resources` struct in case of `<xwhat>` or a `String` in case of `what`. The same applies for `auxwhat`/`xauxwhat`.
Refer to the [serde documentation on that topic](https://serde.rs/string-or-struct.html).

**Comment by jfrimmel** [comment]
This is fixed by #80.
